drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/widgets/ScenarioGrid.java

```diff
@@ -0,0 +1,75 @@
+ * limitations under the License.
+ */
+package org.drools.workbench.screens.scenariosimulation.client.widgets;
+
+import java.util.Map;
+
+import com.ait.lienzo.client.core.event.NodeMouseDoubleClickHandler;
+import com.ait.lienzo.shared.core.types.EventPropagationMode;
+import org.drools.workbench.screens.scenariosimulation.client.handlers.ScenarioSimulationGridPanelDoubleClickHandler;
+import org.drools.workbench.screens.scenariosimulation.client.models.ScenarioGridModel;
+import org.drools.workbench.screens.scenariosimulation.client.renderers.ScenarioGridRenderer;
+import org.drools.workbench.screens.scenariosimulation.client.values.ScenarioGridCellValue;
+import org.uberfire.ext.wires.core.grids.client.widget.grid.impl.BaseGridWidget;
+import org.uberfire.ext.wires.core.grids.client.widget.layer.GridSelectionManager;
+import org.uberfire.ext.wires.core.grids.client.widget.layer.pinning.GridPinnedModeManager;
+
+import static org.drools.workbench.screens.scenariosimulation.client.utils.ScenarioSimulationUtils.getScenarioGridColumn;
+
+public class ScenarioGrid extends BaseGridWidget {
+
+    private final ScenarioGridLayer scenarioGridLayer;
+    private final ScenarioGridPanel scenarioGridPanel;
+
+    public ScenarioGrid(ScenarioGridModel model, ScenarioGridLayer scenarioGridLayer, ScenarioGridRenderer renderer, ScenarioGridPanel scenarioGridPanel) {
+        super(model, scenarioGridLayer, scenarioGridLayer, renderer);
+        this.scenarioGridLayer = scenarioGridLayer;
+        this.scenarioGridPanel = scenarioGridPanel;
+        setEventPropagationMode(EventPropagationMode.NO_ANCESTORS);
+    }
+
+    @Override
+    protected NodeMouseDoubleClickHandler getGridMouseDoubleClickHandler(final GridSelectionManager selectionManager,
+                                                                         final GridPinnedModeManager pinnedModeManager) {
+        return new ScenarioSimulationGridPanelDoubleClickHandler(this,
+                                                                 selectionManager,
+                                                                 pinnedModeManager,
+                                                                 renderer);
+    }
+
+    public void setContent(Map<Integer, String> headersMap, Map<Integer, Map<Integer, String>> rowsMap) {
+        ((ScenarioGridModel) model).clear();
+        ((ScenarioGridModel) model).bindContent(headersMap, rowsMap);
+        setHeaderColumns(headersMap);
+        appendRows(rowsMap);
+    }
+
+    private void setHeaderColumns(Map<Integer, String> headersMap) {
+        headersMap.forEach((columnIndex, columnTitle) ->
+                                   model.insertColumn(columnIndex, getScenarioGridColumn(columnTitle, scenarioGridPanel, scenarioGridLayer)));
+
+
+    }
+
+    private void appendRows(Map<Integer, Map<Integer, String>> rowsMap) {
+        if (!rowsMap.isEmpty()) {
+            rowsMap.forEach((rowIndex, cellValueMap) -> {
```

Comment:
I believe using `forEach` ensures nothing is done if collection is empty. We should remove the outer `if`.

Reply:
Thanks!